src/status_im/common/biometric/events.cljs

```diff
@@ -1,120 +1,128 @@
-
-(defn get-icon-by-type
-  [biometric-type]
-  (condp = biometric-type
-    constants/biometrics-type-face-id :i/face-id
-    :i/touch-id))
-
-(re-frame/reg-fx
- :biometric/get-supported-biometric-type
+(rf/reg-fx
+ :biometric/get-supported-type
  (fn []
    ;;NOTE: if we can't save user password, we can't use biometric
    (keychain/can-save-user-password?
     (fn [can-save?]
-      (when (and can-save? (not android-device-blacklisted?))
+      (when (and can-save? (not utils/android-device-blacklisted?))
         (-> (biometrics/get-supported-type)
             (.then (fn [type]
-                     (rf/dispatch [:biometric/get-supported-biometric-type-success type])))))))))
+                     (rf/dispatch [:biometric/set-supported-type type])))))))))
+
+(defn set-supported-type
+  [{:keys [db]} [supported-type]]
+  {:db (assoc-in db [:biometrics :supported-type] supported-type)})
 
-(rf/defn get-supported-biometric-auth-success
-  {:events [:biometric/get-supported-biometric-type-success]}
-  [{:keys [db]} supported-type]
-  {:db (assoc db :biometric/supported-type supported-type)})
+(schema/=> set-supported-type events-schema/?set-supported-type)
+(rf/reg-event-fx :biometric/set-supported-type set-supported-type)
 
-(rf/defn show-message
-  {:events [:biometric/show-message]}
-  [_ error]
-  (let [code    (ex-cause error)
-        content (if (#{::biometrics/not-enrolled
-                       ::biometrics/not-available}
+(defn show-message
+  [_ [code]]
+  (let [content (if (#{:biometrics/not-enrolled-error
+                       :biometrics/not-available-error}
                      code)
                   (i18n/label :t/grant-face-id-permissions)
                   (i18n/label :t/biometric-auth-error {:code code}))]
-    {:effects.utils/show-popup
-     {:title   (i18n/label :t/biometric-auth-login-error-title)
-      :content content}}))
+    {:fx [[:effects.utils/show-popup
+           {:title   (i18n/label :t/biometric-auth-login-error-title)
+            :content content}]]}))
 
+(schema/=> show-message events-schema/?show-message)
+(rf/reg-event-fx :biometric/show-message show-message)
 
-(re-frame/reg-fx
+(rf/reg-fx
  :biometric/authenticate
- (fn [{:keys [on-success on-fail prompt-message]}]
+ (fn [{:keys [prompt-message on-success on-fail on-cancel on-done]
+       :or   {on-done    identity
+              on-success identity
+              on-cancel  identity
+              on-fail    identity}}]
    (-> (biometrics/authenticate
         {:prompt-message          (or prompt-message (i18n/label :t/biometric-auth-reason-login))
          :fallback-prompt-message (i18n/label
                                    :t/biometric-auth-login-ios-fallback-label)
          :cancel-button-text      (i18n/label :t/cancel)})
        (.then (fn [not-canceled?]
-                (when (and on-success not-canceled?)
-                  (on-success))))
+                (on-done)
+                (if not-canceled?
```

Comment:
if successful?

Reply:
so here it can only be successful or canceled, but not failed. I thought naming it `not-canceled?` would make this distinction clearer, cause it looked a bit weird like this:

```clj
(if successful?
    (on-success)
    (on-cancel))
```

looking at it now it seems fine, so maybe it was a case of overthinking 😆

Reply:
changed to your suggestion 👍